### ISSUE TITLE
Android support

### ISSUE THREAD

**Issue by samhunta** [tomauty/react-native-chart#59]
Does this work for Android?

**Comment by shanemcgraw** [comment]
According to https://github.com/tomauty/react-native-chart/issues/58, no.

**Comment by tomauty** [comment]
I'll leave this one open. I'm starting work on it next week. It will be under the `2.0` branch, since this will be a massive overhaul.

**Comment by pinhopro** [comment]
@tomauty  Please, post your bitcoin wallet.  I will send you a donation to help you with the android port :)

**Comment by tomauty** [comment]
hah don't worry about that, I'll be trying to figure it out soon :)

**Comment by cablegunmaster** [comment]
How is this coming along? 

Plus 1 for the effort, its one of the best react-native graphs/charts thusfar!
 :+1:

**Comment by tomauty** [comment]
I'm working on a 2.0 branch that will be pure JS. Hope to make some progress this week.

**Comment by holyxiaoxin** [comment]
I will donate as well! <3

**Comment by Rpaixao** [comment]
I really need this component! Go ahead! I am able to donate too ;)

**Comment by tomauty** [comment]
In lieu of donations, I'd appreciate links to other react charting libraries with a similar feature set. Looking for some guidance on architecture. I'm hoping to have a useable bar chart on 2.0 next week, then line, then feature parity.

**Comment by cablegunmaster** [comment]
Keep up the goodwork Tomauty, tried to test it today but guess the 2.0 version is not ready yet ;).

**Comment by tomauty** [comment]
Hey y'all. Some exciting news on this front. I've had the go-ahead at work to go full-force with this for part of the week. Here are some screenshots of 2.0 so far.

![simulator screen shot apr 26 2016 10 12 33 am](https://cloud.githubusercontent.com/assets/1112872/14821098/6d7711d4-0b97-11e6-9e09-2c8572a3d127.png)
![simulator screen shot apr 26 2016 10 12 35 am](https://cloud.githubusercontent.com/assets/1112872/14821099/6d7dd7a8-0b97-11e6-828f-c66b095e0d3e.png)

Unfortunately there are some difficulties with animations which might make 2.0 an alpha/beta for a while, but it should be usable in its alpha stage on both platforms sometime next week for testing.

**Comment by tomauty** [comment]
Note that 2.0 has no objective-C code, or dependencies :)

**Comment by tofrookie** [comment]
i can't wait :)

**Comment by pinhopro** [comment]
Awesome ....   👍  post a bitcoin address so I can give you a tip :)

**Comment by cablegunmaster** [comment]
Keep up the good work Tomauty! :+1:

**Comment by tomauty** [comment]
Update – haven't had much time to work on this the past week or two, but I will have lots of time this weekend. I will update this thread when 2.0 is stable enough so you guys can test. Tentative plan is to merge 2.0 into master branch on Sunday, with approximate feature parity.

**Comment by rgoldiez** [comment]
Any update?

**Comment by tomauty** [comment]
Sorry, got caught up a bit. Almost done, just fixing some pie chart bugs and standardizing the API :)

**Comment by rupeshdeotale97** [comment]
@tomauty  Any update for android RN 0.24 .

**Comment by rgoldiez** [comment]
Any update on 2.0?

**Comment by tomauty** [comment]
Sorry I'm way behind schedule here. Things have slowed down a bit at work, so I'll try my best to get something out the door this week. Please bug me about it :)

**Comment by cablegunmaster** [comment]
Ah, easy is it working??  _bugs you about it ;) &_ just because you asked.

**Comment by Mindaugas-Jacionis** [comment]
Any news? :)

**Comment by tomauty** [comment]
Sorry I'm posting in a bunch of issues. I'm standardizing the API and documentation today, then it should be ready for use.

**Comment by cablegunmaster** [comment]
I am awaiting eagerly hopefully to put this in production after release.

**Comment by cablegunmaster** [comment]
@tomauty 

Feedback on current stage of production:

what I did to make it run changed the dependency's on the react and react-native because of some bug which didnt make it run.  -> https://github.com/jhabdas/react-native-webpack-starter-kit/issues/101

Package.json:
`
  "dependencies": {
    "react": "^15.1.0",
    "react-native": "^0.27.2"
  },
`

Index.android.js:
`
   import Chart from './src/Chart'; 
` 
`
<Chart  
showAxis 
type='bar'

```
                    data={[[0, 1], [1, 3],[3, 7],[4, 9],[4, 200],[5, 20],[6, 60]]}
                    style={styles.chart}
                    axisColor={colors.grey}
                    gridColor={colors.grey}
                    chartData={this.state.barChart}
                    xAxisLabels={this.state.xLabels}
                    axisLabelColor={colors.grey}
                    showYAxisLabels
                    showXAxisLabels
                />
```

`
Barchart seems stable so far, as far as the other 2 charts, the  pie chart and the line chart, its a no show yet. No idea what pi chart is doing but it ain't doing as its supposed to.

In Short:
- Updated dependancy react-js natve to show the chart. (couldnt understand the error)
- Still a bug in receiving the props of the 'type' ( if you wanted to not include the 'type'  in Chart itself. ) 
- A bug in the piechart which overflows it with one color.
- Linechart not showing (yet).

Hopefully this will help your development ;)! Trying to help you.

**Comment by tomauty** [comment]
Sorry guys, I am completely swamped by work this week :(

**Comment by tomauty** [comment]
This was on android, yes?

**Comment by cablegunmaster** [comment]
Correct this was on android

**Comment by Mindaugas-Jacionis** [comment]
Hi! How is it going with 2.0? :)

**Comment by tomauty** [comment]
This week. I promise! :)

**Comment by esutton** [comment]
Links to a **cross-platform** charting library that is reported to be releasing the react-native version this Friday, June 17.  
- http://formidable.com/open-source/victory/
- https://github.com/FormidableLabs/victory/blob/master/ROADMAP.md
- [Cross-platform charting with React & React Native by Ken Wheeler at react-europe 2016](https://www.youtube.com/watch?v=gt0VtewBZ94)

Documentation looks good as well as the sample images.

No idea how stable or usable it will be from the start but they look like they have funding and full-time resources.

[Many Examples:](http://formidable.com/open-source/victory/)

<img width="634" alt="screen shot 2016-06-15 at 5 10 13 pm" src="https://cloud.githubusercontent.com/assets/6115047/16099111/23ab54dc-331c-11e6-8570-58b6add45c7e.png">

**Comment by tomauty** [comment]
https://github.com/facebook/react-native/commit/13acd7ed7e641f80c8cfecf949a96c336300d403

This is causing the stagnant pie chart issue. Any workarounds or suggestions would be appreciated.

**Comment by ggaabe** [comment]
@esutton Where did you see that this library would be available by June 17th?

**Comment by esutton** [comment]
A Victory developer on Gitter told me so.

Sent from my iPhone

On Jun 17, 2016, at 12:44 PM, Gabriel Garrett <notifications@github.com<mailto:notifications@github.com>> wrote:

@esuttonhttps://github.com/esutton Where did you see that this library would be available by June 17th?

##

**Comment by rgoldiez** [comment]
Almost, their devs responded to a github issue in their repo saying this week.

**Comment by tomauty** [comment]
Hi guys. Fingers crossed here. I will appreciate lots of feedback and PRs in the coming weeks. I've published on npm as `1.0.0-beta`.

**Comment by tomauty** [comment]
With that in mind, I will close this for now and accept more specific issues regarding Android support.

**Comment by Dacello** [comment]
@tomauty how do you go about linking libraries for android in 1.0.0-beta version?

**Comment by tomauty** [comment]
It should be linked automatically for Android. iOS requires ART linking.

**Comment by Dacello** [comment]
Ah ok. Thanks for the quick reply
